File: LAB3/C#/Lab3POP/Lab3POP/Storage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Lab3POP
{
    internal class Storage
    {
        public Semaphore access;
        public Semaphore full;
        public Semaphore empty;

        private List<string> _storage = new List<string>();

        public Storage(int storageSize)
        {
            access = new Semaphore(1, 1);
            full = new Semaphore(storageSize, storageSize);
            empty = new Semaphore(0,storageSize);
        }

        public string GetItem()
        {
            return _storage.ElementAt(0);
        }

        public void AddItem(string item)
        {
            _storage.Add(item);
        }

        public void RemoveItem()
        {
            _storage.RemoveAt(0);
        }
    }
}

File: LAB3/JAVA/LAB3POP/src/Storage.java
import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.concurrent.Semaphore;

public class Storage {

    public Semaphore access;
    public Semaphore full;
    public Semaphore empty;

    private ArrayList<String> _storage = new ArrayList<>();

    public Storage(int size){
        access= new Semaphore(1);
        full = new Semaphore(size);
        empty = new Semaphore(0);
    }

    public String getItem()
    {
        return _storage.get(0);
    }

    public void addItem(String item)
    {
        _storage.add(item);
    }

    public void removeItem()
    {
        _storage.remove(0);
    }
}
